### DISCUSSION TITLE
bluetooth is not working

### DISCUSSION THREAD

**Discussion by ShamanK18** [blueman-project/blueman#2241] [Q&A]
blueman:
BlueZ:
Distribution:
Desktop environment:

* [ ] I have consulted the [Troubleshooting page](https://github.com/blueman-project/blueman/wiki/Troubleshooting) and done my best effort to follow.

something is not working in my bluetooth as you can see that even after the search option no device 
![Screenshot_2023-12-26_18-39-30](https://github.com/blueman-project/blueman/assets/131748408/a50acc8d-2ff1-4567-a4b9-72a66d0e1c99)
![Screenshot_2023-12-26_18-40-00](https://github.com/blueman-project/blueman/assets/131748408/49943149-3c5b-472a-92b6-2db0891557d1)

**Comment by cschramm** [comment]
Assuming that you're able to discover some devices from another device at the same time, check if it might be hidden due to no name provided in the "View" menu.

**Comment by ShamanK18** [comment]
> Assuming that you're able to discover some devices from another device at the same time, check if it might be hidden due to no name provided in the "View" menu.

I checked everything sir but no device is showing nor is the name of the PC showing in the phone.

**Comment by infirit** [comment]
Iirc, the Bluetooth adapter is hidden by default, open blueman-adapters to change this.

**Comment by ShamanK18** [comment]
> Iirc, the Bluetooth adapter is hidden by default, open blueman-adapters to change this.

I didn't see anything else other than the screenshots I have given above.

**Comment by ShamanK18** [comment]
> Iirc, the Bluetooth adapter is hidden by default, open blueman-adapters to change this.

I have checked all the settings

**Comment by cschramm** [comment]
Not sure I can follow. You have an unknown version of blueman running on an unknown system and try to discover another PC? You do see that PC with another device that it's not paired with - say, your smartphone -, so it is discoverable, but you do not see it in blueman on the unknown system? Are you able to discover other devices - say, your smartphone? Do you see discoverable devices in other tools on the unknown system - say, bluetoothctl?

**Comment by ShamanK18** [comment]
> Not sure I can follow. You have an unknown version of blueman running on an unknown system and try to discover another PC? You do see that PC with another device that it's not paired with - say, your smartphone -, so it is discoverable, but you do not see it in blueman on the unknown system? Are you able to discover other devices - say, your smartphone? Do you see discoverable devices in other tools on the unknown system - say, bluetoothctl?

Check this, whatever you want to check, check it and then suggest a solution to fix it.[
![20240106_170803](https://github.com/blueman-project/blueman/assets/131748408/2d3c0f6b-7c54-44b1-9589-2d417a76744e)
](url
![Uploading 20240106_170755.jpg…](
![20240106_170748](https://github.com/blueman-project/blueman/assets/131748408/f0e4892e-99e2-460c-af60-964efb8813f2)
![20240106_170740](https://github.com/blueman-project/blueman/assets/131748408/932c4db2-cb11-4f15-9351-132fc71bdd00)
![20240106_170734](https://github.com/blueman-project/blueman/assets/131748408/1ff6374c-fdf5-4e4a-ab56-34ba5fa2fb34)
![20240106_170722](https://github.com/blueman-project/blueman/assets/131748408/85112d81-388e-4420-beb2-da9f2291aaa1)
![20240106_170701](https://github.com/blueman-project/blueman/assets/131748408/308dbb20-c184-4ae1-a04f-64cd89708a4d)
![20240106_170653](https://github.com/blueman-project/blueman/assets/131748408/729b20d7-ec1a-4f20-87b1-360f7e20996d)

**Comment by cschramm** [comment]
I cannot conclude anything I asked for from those photos, just that it seems to be some blueman 2.3, that you are hiding unnamed devices and that the adapter is set discoverable.

I'm still unsure if you're trying to discover some device in blueman or to make the system discoverable with blueman. The latter seems to work fine. If you cannot actually discover it, double check the procedure and possibly try from a different device. In case of issues with your adapter, I'd expect some kernel logs.

**Comment by ShamanK18** [comment]
> I cannot conclude anything I asked for from those photos, just that it seems to be some blueman 2.3, that you are hiding unnamed devices and that the adapter is set discoverable.
> 
> I'm still unsure if you're trying to discover some device in blueman or to make the system discoverable with blueman. The latter seems to work fine. If you cannot actually discover it, double check the procedure and possibly try from a different device. In case of issues with your adapter, I'd expect some kernel logs.

I have to connect bluetooth ear buds to it please help how can I fix it

> I cannot conclude anything I asked for from those photos, just that it seems to be some blueman 2.3, that you are hiding unnamed devices and that the adapter is set discoverable.
> 
> I'm still unsure if you're trying to discover some device in blueman or to make the system discoverable with blueman. The latter seems to work fine. If you cannot actually discover it, double check the procedure and possibly try from a different device. In case of issues with your adapter, I'd expect some kernel logs.

**Comment by cschramm** [comment]
So you want to discover ear buds from a system that runs blueman 2.3 (I'm ignoring that you talked about a PC not being visible on a phone :shrug:).

First check that the ear buds are actually discoverable. You can use your phone to search for them (make sure they are not already known to the phone initially). If you cannot find them there either, the conclusion is that they are not discoverable. Consult their manual or support on how to get them paired.

In case they do show up on the phone but not in blueman when searching at the same time and distance, run `bluetoothctl` in a terminal window on the system that you're running blueman on, enter `scan on` in it and see if they do show up there. If not, you successfully ruled out an issue with blueman. In case no devices show up at all, you might have a problem with the system's adapter or driver and there might be kernel logs regarding it.

**Comment by ShamanK18** [comment]
> So you want to discover ear buds from a system that runs blueman 2.3 (I'm ignoring that you talked about a PC not being visible on a phone 🤷).
> 
> First check that the ear buds are actually discoverable. You can use your phone to search for them (make sure they are not already known to the phone initially). If you cannot find them there either, the conclusion is that they are not discoverable. Consult their manual or support on how to get them paired.
> 
> In case they do show up on the phone but not in blueman when searching at the same time and distance, run `bluetoothctl` in a terminal window on the system that you're running blueman on, enter `scan on` in it and see if they do show up there. If not, you successfully ruled out an issue with blueman. In case no devices show up at all, you might have a problem with the system's adapter or driver and there might be kernel logs regarding it.

I have tried all this but it is not showing anything, neither is the name of bluetooth of my phone nor of ear buds shown but it works well in windows but there also after updating the bluetooth driver, bluetooth is not working. It is working but nothing is working here nor did I find the option to update the bluetooth driver anywhere.

**Comment by cschramm** [comment]
So you can discover the ear buds from a Windows system but at the same time neither from the system running blueman nor from some phone? The ear buds might be coupled to the Windows system then, although the Windows system does not have a pairing anymore. Consult their manual or support on how to get them decoupled.

**Reply by ShamanK18** [reply]
It gets connected to the phone but what is the problem with linux mint itself, windows does not work well in my laptop, it runs very stuck but it gets connected via bluetooth but everything else is fine in it, only bluetooth. not working please give any solution

**Comment by GentlemanHu** [comment]
In my case, I changed my card to broadcom card for hackintosh, so it seems missing firmware.

> yay -S broadcom-bt-firmware

then reboot, you good to go.

https://forum.manjaro.org/t/bluetooth-doesnt-detect-any-devices/105739

**Reply by ShamanK18** [reply]
Not working this error is showing
![20240117_192019](https://github.com/blueman-project/blueman/assets/131748408/f478d85f-15fe-49f4-b8ce-c903f6127afb)
 any other solution

**Reply by GentlemanHu** [reply]
What is your os? I'm using arch linux, so I use yay to intall package; 

Possible solution:

1. Try google and find a way to know what your WiFi card is.
2. If it is broadcom card, then try following step

Try google how to install `broadcom-bt-firmware` or just `apt install -y broadcom-bt-firmware`   or search `broadcom-bt-firmware` package for your system then try to install it.

**Reply by ShamanK18** [reply]
I am using linux mint

**Comment by ShamanK18** [comment]
My problem is not solved yet
